### PR TITLE
Timur katimull 21 30 tasks

### PR DESCRIPTION
**Pull request by linkouth**
Adds solution 21-30 tasks

### CODE REVIEW

File: tasks/29_remove-duplicates/remove-duplicates.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/**
+ * removeDuplicates - removes duplicates from an array
+ * of numbers and returns it as a result
+ *
+ * @param {Array<Number>} a
+ * @return {Array<Number>}
+ */
+export function removeDuplicates(a) {
+  return a.filter((value, index) => a.indexOf(value) === index);
```
Comment (alexkond) [inline tasks/29_remove-duplicates/remove-duplicates.js:9]: StackOverflow solution 😜

**Review by alexkond — DISMISSED** [review]

**Review by vovcyan — APPROVED** [review]

File: tasks/21_check-letters/check-letters.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * checkLetters - returns true if the string in the first
+ * element of the array contains all of the letters of
+ * the string in the second element of the array
+ *
+ * @param {Array<String>} arr
+ * @return {Boolean}
+ */
+export function checkLetters(arr) {
+  let firstString = arr[0].toLowerCase();
```
Comment (vovcyan) [inline tasks/21_check-letters/check-letters.js]: const

File: tasks/21_check-letters/check-letters.spec.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-// import { checkLetters } from './check-letters';
+import { checkLetters } from './check-letters';
 
-xdescribe('check-letters', () => {
```
Comment (vovcyan) [inline tasks/21_check-letters/check-letters.spec.js]: Please do not commit files with tests

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/21_check-letters/check-letters.js (hunk 2 above)
Comment (linkouth) [reply tasks/21_check-letters/check-letters.js]: Fixed

File: tasks/21_check-letters/check-letters.spec.js (hunk 3 above)
Comment (linkouth) [reply tasks/21_check-letters/check-letters.spec.js]: Reverted

File: tasks/27_basic-subclasses/basic-subclasses.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+export class God {
+  /**
+   * @return {Array<Human>}
+   */
+  static create() {
+    return [new Man('Adam'), new Woman('Eve')];
```
Comment (vovcyan) [inline tasks/27_basic-subclasses/basic-subclasses.js]: const man = new Man('Adam');
const woman = new Woman('Eve');

return [man, woman];

**Review by vovcyan — CHANGES_REQUESTED** [review]

File: tasks/27_basic-subclasses/basic-subclasses.js (hunk 4 above)
Comment (linkouth) [reply tasks/27_basic-subclasses/basic-subclasses.js]: Fixed

File: tasks/24_index-to-insert/index-to-insert.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+/**
+ * getIndexToInsert - returns the lowest index at which
+ * a value should be inserted into an array once it has been sorted
+ *
+ * @param {Array<Number>} arr
+ * @param {Number} num
+ * @return {Number}
+ */
+export function getIndexToInsert(arr, num) {
+  const sortedArray = arr.sort((a, b) => a - b);
```
Comment (KalachevDev) [inline tasks/24_index-to-insert/index-to-insert.js:10]: I believe it works this way by default so no need to pass function into sort

**Review by KalachevDev — APPROVED** [review]

File: tasks/24_index-to-insert/index-to-insert.js (hunk 5 above)
Comment (linkouth) [reply tasks/24_index-to-insert/index-to-insert.js:10]: By default sort() compare elements of the array as String. If we want correct sort array of numbers we should pass function into sort.

**Review by savchukm — APPROVED** [review]

**Review by alexkond — APPROVED** [review]

**Pull request merged**